openverse_catalog/dags/common/sensors.py

```diff
@@ -0,0 +1,93 @@
+import os
+from datetime import timedelta
+from typing import Iterable
+
+from airflow.exceptions import AirflowException
+from airflow.models import DagModel, DagRun
+from airflow.sensors.base import BaseSensorOperator
+from airflow.triggers.temporal import TimeDeltaTrigger
+from airflow.utils.session import provide_session
+from airflow.utils.state import State
+from sqlalchemy import func
+
+
+class ExternalDAGsSensorAsync(BaseSensorOperator):
+    """
+    Waits for a list of different DAGs to not be running, freeing up
+    a worker slot while it's waiting.
+
+    external_dag_ids: A list of dag_ids that you want to wait for
+    :param check_existence: Set to `True` to check if the external task exists (when
+        external_task_id is not None) or check if the DAG to wait for exists (when
+        external_task_id is None), and immediately cease waiting if the external task
+        or DAG does not exist (default value: False).
+    """
+
+    # TODO: Not sure if this is necessary
+    template_fields = ["external_dag_ids"]
+
+    # TODO: I don't understand what operator extra links are, and whether it's
+    # necessary to implement this.
+    # @property
+    # def operator_extra_links(self):
+    #     """Return operator extra links"""
+    #     return [ExternalDAGsSensorLink()]
```

Comment:
We certainly could, but I think it would just point to the other data refresh DAGs :thinking: might be a bit redundant, since each one will point to all the others?